pkg/ebpf/debugfs.go

```diff
@@ -0,0 +1,73 @@
+package ebpf
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/pkg/errors"
+)
+
+type kprobeStats struct {
+	hits int64
+	miss int64
+}
+
+// KPROBEPROFILE is the default path to the kprobe_profile file
+const KPROBEPROFILE = "/sys/kernel/debug/tracing/kprobe_profile"
+
+// GetProbeStats gathers stats about the # of kprobes triggered /missed by reading the kprobe_profile file
+func GetProbeStats() map[string]int64 {
+	m, err := readKprobeProfile(KPROBEPROFILE)
+	if err != nil {
+		log.Debugf("error retrieving probe stats: %s", err)
+		return map[string]int64{}
+	}
+
+	res := make(map[string]int64, 2*len(m))
+	for event, st := range m {
+		res[fmt.Sprintf("%s_hits", event)] = st.hits
+		res[fmt.Sprintf("%s_misses", event)] = st.miss
+	}
+
+	return res
+}
+
+// readKprobeProfile reads a /sys/kernel/debug/tracing/kprobe_profile file and returns a map of probe -> stats
+func readKprobeProfile(path string) (map[string]kprobeStats, error) {
+	f, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading kprobe profile file at: %s", path)
+	}
+
+	lines := strings.Split(string(f), "\n")
```

Comment:
You don't have to do this, but I'm wondering if using `bufio.Scanner` would save some memory: https://golang.org/pkg/bufio/#Scanner

Reply:
Done 👍